components/script/dom/canvasrenderingcontext2d.rs

```diff
@@ -1071,10 +1070,12 @@ impl CanvasRenderingContext2DMethods for CanvasRenderingContext2D {
 
         // Un-premultiply alpha
         for chunk in data.chunks_mut(4) {
-            let alpha = chunk[3] as usize;
-            chunk[0] = UNPREMULTIPLY_TABLE[256 * alpha + chunk[0] as usize];
-            chunk[1] = UNPREMULTIPLY_TABLE[256 * alpha + chunk[1] as usize];
-            chunk[2] = UNPREMULTIPLY_TABLE[256 * alpha + chunk[2] as usize];
+            let alpha = chunk[3] as u16;
+            if alpha != 0 {
+                chunk[0] = ((chunk[0] as u16 * 255 + (alpha / 2)) / alpha) as u8;
```

Comment:
I took this from I've been doing a lot of Googling trying to figure out what the "right" answer is here. https://bugzilla.mozilla.org/show_bug.cgi?id=659725#c31, not sure what the best solution here is though.